frontend/fingertips-frontend/lib/fetchData.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  Configuration,
+  WeatherForecast,
+  WeatherForecastApi,
+} from '@/generated-sources/api-client';
+import { connection } from 'next/server';
+
+export async function fetchData(): Promise<WeatherForecast[]> {
```

Comment:
I don't think we're going to be able to have a generic `fetchData` function as we expand the application to call multiple API endpoints with different parameters. Perhaps what would be useful is keeping this function as something that reads the API URL environment variable and creates (and returns) a `Configuration` object, and then we can leave the part that actually uses the `Configuration` to call the specific API in the page?